### DISCUSSION TITLE
Interested to contribute to the Calyx backend in Dahlia

### DISCUSSION THREAD

**Discussion by Mark1626** [cucapra/futil#1886] [Introductions]
Hi everyone,

# Intro

I am a research engineer, my work involves HPC and Accelerated Computing. I have been using Dahlia recently, I found the HLS backend quite productive. I'm interested in MLIR CIRCT and Calyx, so I'm currently starting to explore lowering Dahlia to Calyx. In the Calyx documentation it is mention that the Dahlia backend is not complete.

I would like to contribute to Calyx backend, can you let me know where I can get started.

## Some things I currently noticed and planned to contribute to

1. Documentation around the backend as well as lowering
2. Lowering Dahlia Record to Calyx

**Comment by rachitnigam** [comment]
Hi @Mark1626! Thank you for contributing to Dahlia and we're super glad you've found it to be useful! The Calyx backend for Dahlia is reasonably mature and can compile [polybench benchmark](https://github.com/cucapra/calyx-evaluation) for example. The development of this backend has slowed down because we've been thinking of designing a new HLS frontend (#1680). 

Regardless, if you're excited about building upon the Dahlia backend, we'd love to help you. We started a new, public [zulip channel for Calyx](https://calyx.zulipchat.com/join/w7prpemmpdhqfn6j767otkme/) that you should consider joining! We can provide answers more quickly if you have question on it!
